Makefile

```diff
@@ -176,17 +180,19 @@ endif
 $(ROM): $(ELF)
 	$(ELF2ROM) -cic 6105 $< $@
 
-$(ELF): $(TEXTURE_FILES_OUT) $(ASSET_FILES_OUT) $(O_FILES) build/ldscript.txt build/undefined_syms.txt
+$(ELF): $(TEXTURE_FILES_OUT) $(ASSET_FILES_OUT) $(O_FILES) $(OVL_RELOC_FILES) build/ldscript.txt build/undefined_syms.txt
 	$(LD) -T build/undefined_syms.txt -T build/ldscript.txt --no-check-sections --accept-unknown-input-arch --emit-relocs -Map build/z64.map -o $@
 
+$(OVL_RELOC_FILES): | $(O_FILES)
```

Comment:
This rule is there to ensure the correct order between `OVL_RELOC_FILES` and `O_FILES` (since the reloc files need all their corresponding files to be compiled first), so it has to be there on its own

Reply:
To be clear, currently is:

```mk
OVL_RELOC_FILES := $(shell grep -o '[^"]*_reloc.o' spec )

(OVL_RELOC_FILES): | $(O_FILES)

build/src/overlays/%_reloc.o: build/spec
	$(FADO) $$(tools/reloc_preq $< $(notdir $(@D))) -n $(notdir $(@D)) -o $(@:.o=.s) -M $(@:.o=.d)
	$(AS) $(ASFLAGS) $(@:.o=.s) -o $@
```

, is this different from:

```mk
OVL_RELOC_FILES := $(shell grep -o '[^"]*_reloc.o' spec )

build/src/overlays/%_reloc.o: build/spec | $(O_FILES)
	$(FADO) $$(tools/reloc_preq $< $(notdir $(@D))) -n $(notdir $(@D)) -o $(@:.o=.s) -M $(@:.o=.d)
	$(AS) $(ASFLAGS) $(@:.o=.s) -o $@
```

?

Reply:
I'm not sure about this, to be honest, the make docs don't seem very clear on what happens when you have two rules of probably equal scope. Roman's idea makes sense, though, although I can never remember quite how `.PHONY` works.

Reply:
To me it sounds equivalent. At the very least the two rules should be near each other imo

Reply:
I'm not sure how we want the makefile organised, currently all the phony targets are in the same section. I can move it, just not sure it fits with the pattern rules to do so. @Roman971 , would you want to change other references to `O_FILES` to use `o_files` as well, like  `$(ELF)`?